osgp/platform/osgp-adapter-domain-tariffswitching/src/test/java/org/opensmartgridplatform/adapter/domain/tariffswitching/application/services/ScheduleManagementServiceTest.java

```diff
@@ -0,0 +1,130 @@
+	private CorrelationIds correlationIds;
+	
+	@InjectMocks
+	private ScheduleManagementService scheduleManagementService;
+	
+	private List<ScheduleEntry> scheduleEntries;
+	private List<DeviceOutputSetting> deviceOutputSettings;
+	private List<LightValue> lightValues;
+	private final long scheduleTime = (long) 0.0f;
+	private final String messageType = "testType";
+	private final int priority = 1;
+
+	public void setup() throws FunctionalException {
+		this.scheduleEntries = new ArrayList<>();
+		this.deviceOutputSettings = new ArrayList<>();
+		this.lightValues = new ArrayList<>();
+
+		final DeviceOutputSetting dos = Mockito.mock(DeviceOutputSetting.class);
+		when(dos.getOutputType()).thenReturn(RelayType.TARIFF_REVERSED);
+		
+		final ScheduleEntry entry = Mockito.mock(ScheduleEntry.class);
+		when(entry.getLightValue()).thenReturn(this.lightValues);
+		
+		final LightValue value = Mockito.mock(LightValue.class);
+		
+		//idk how many entries are typically required
+		for(int i =0; i < 3; i++) {
+			this.scheduleEntries.add(entry);
+			this.deviceOutputSettings.add(dos);
+			this.lightValues.add(value);
+		}
+		
+		when(this.device.getId()).thenReturn((long) 0.0f);
+		when(this.ssld.getOutputSettings()).thenReturn(this.deviceOutputSettings);
+		when(this.ssldRepository.findById(any())).thenReturn(Optional.of(this.ssld));
+		when(this.device.getIpAddress()).thenReturn("127.0.0.1");
+		when(this.deviceDomainService.searchActiveDevice(any(), any(ComponentType.class))).thenReturn(this.device);
+	}
+	
+	@Test
+	public void testSetTariffSchedule() throws FunctionalException {
+		this.setup();
+		when(this.device.getDeviceType()).thenReturn(Ssld.SSLD_TYPE);
+
+		final ArgumentCaptor<RequestMessage> requestMessageArgumentCaptor =
+				ArgumentCaptor.forClass(RequestMessage.class);
+		final ArgumentCaptor<Long> scheduleTimeCaptor = ArgumentCaptor.forClass(long.class);
+		final ArgumentCaptor<String> ipCaptor = ArgumentCaptor.forClass(String.class);
+		final ArgumentCaptor<String> messageTypeCaptor = ArgumentCaptor.forClass(String.class);
+		final ArgumentCaptor<Integer> priorityCapture = ArgumentCaptor.forClass(int.class);
+
+		this.scheduleManagementService.setTariffSchedule(this.correlationIds, this.scheduleEntries, this.scheduleTime,
+				this.messageType, this.priority);
+
+		verify(this.osgpCoreRequestMessageSender).send(requestMessageArgumentCaptor.capture(), messageTypeCaptor.capture(),priorityCapture.capture(),ipCaptor.capture(), scheduleTimeCaptor.capture());
+
+		assertThat(this.messageType).isEqualTo(messageTypeCaptor.getValue());
+		assertThat(this.scheduleTime).isEqualTo(scheduleTimeCaptor.getValue());
+		assertThat(this.priority).isEqualTo(priorityCapture.getValue());
+	}
+	
+	@Test
+	public void testSetTariffScheduleFunctionalExceptionThrown() throws FunctionalException {
+		//incorrect type should return an exception
+		assertThatThrownBy(()-> this.scheduleManagementService.setTariffSchedule(this.correlationIds, this.scheduleEntries,
```

Comment:
when using assertThatExceptionOfType(<Exception.>class).isThrownBy(() -> 
we check the type of exception as well